sqlglot/dialects/materialize.py

```diff
@@ -0,0 +1,102 @@
+            "MAP": lambda self: self._parse_map(),
+        }
+
+        def _parse_list(self) -> exp.List:
+            if self._match(TokenType.L_PAREN):
+                e = self.expression(exp.List, expressions=[self._parse_select()])
+                self._match_r_paren()
+                return e
+
+            if not self._match(TokenType.L_BRACKET):
+                self.raise_error("Expecting [")
+            entries = self._parse_csv(self._parse_conjunction)
+            if not self._match(TokenType.R_BRACKET):
+                self.raise_error("Expecting ]")
+            return self.expression(exp.List, expressions=entries)
+
+        def _parse_map(self) -> exp.ToMap:
+            if self._match(TokenType.L_PAREN):
+                to_map = self.expression(exp.ToMap, this=self._parse_select())
+                self._match_r_paren()
+                return to_map
+
+            if not self._match(TokenType.L_BRACKET):
+                self.raise_error("Expecting [")
+            entries = self._parse_csv(self._parse_map_entry)
+            if not self._match(TokenType.R_BRACKET):
+                self.raise_error("Expecting ]")
+            return self.expression(exp.ToMap, this=self.expression(exp.Struct, expressions=entries))
+
+        def _parse_map_entry(self) -> exp.PropertyEQ | None:
+            key = self._parse_conjunction()
+            if not key:
+                return None
+            if not self._match(TokenType.FARROW):
+                self.raise_error("Expected =>")
+            value = self._parse_conjunction()
+            return self.expression(exp.PropertyEQ, this=key, expression=value)
+
+    class Generator(Postgres.Generator):
+        SUPPORTS_CREATE_TABLE_LIKE = False
+
+        TRANSFORMS = {
+            **Postgres.Generator.TRANSFORMS,
+            exp.AutoIncrementColumnConstraint: lambda self, e: "",
+            exp.Create: preprocess(
+                [
+                    remove_unique_constraints,
+                    ctas_with_tmp_tables_to_create_tmp_view,
+                ]
+            ),
+            exp.DataType: lambda self, e: self._datatype_sql(e),
+            exp.GeneratedAsIdentityColumnConstraint: lambda self, e: "",
+            exp.OnConflict: lambda self, e: "",
+            exp.PrimaryKeyColumnConstraint: lambda self, e: "",
+            exp.List: lambda self, e: self._list_sql(e),
+            exp.ToMap: lambda self, e: self._to_map_sql(e),
+        }
+
+        def _datatype_sql(self: Materialize.Generator, e: exp.DataType) -> str:
+            if e.is_type("list"):
+                if e.expressions:
+                    return f"{self.expressions(e, flat=True)} LIST"
+                return "LIST"
+            elif e.is_type("map"):
+                if e.expressions:
+                    key, value = e.expressions
+                    return f"MAP[{self.sql(key)} => {self.sql(value)}]"
+            return Postgres.Generator.TRANSFORMS[exp.DataType](self, e)
```

Comment:
Yeah, let's get rid of `exp.List`, `exp.ToMap` and `exp.DataType` from `TRANSFORMS` here and define the corresponding `list_sql`, `tomap_sql` and `datatype_sql` methods. We should also get rid of the `exp.DataType` method in Postgres like Vaggelis suggested and create the corresponding method there too.

Reply:
Hey 👋 

Thank you for all this! I just went through all of the change requests except this one. Will look into it shortly.

Reply:
>  I think it would be better if Postgres moved _datatype_sql to be a function ofPostgres.Generator, then you could do super().datatype_sql() here afaict

Just did that, let me know if this is what you had in mind!